qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,169 @@
+from qiskit_experiments import AnalysisResult
+from .analysis_functions import exp_fit_fun, curve_fit_wrapper
+
+
+class T1Analysis(BaseAnalysis):
+    """T1 Experiment result analysis class."""
+
+    # pylint: disable=arguments-differ, unused-argument
+    def _run_analysis(
+        self, experiment_data, t1_guess=None, amplitude_guess=None, offset_guess=None, **kwargs
+    ) -> Tuple[AnalysisResult, None]:
+        """
+        Calculate T1
+
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze
+            t1_guess (float): Optional, an initial guess of T1
+            amplitude_guess (float): Optional, an initial guess of the coefficient of the exponent
+            offset_guess (float): Optional, an initial guess of the offset
+            kwargs: Trailing unused function parameters
+
+        Returns:
+            The analysis result with the estimated T1
+        """
+
+        size = len(experiment_data._data)
+        delays = np.zeros(size, dtype=float)
+        means = np.zeros(size, dtype=float)
+        stddevs = np.zeros(size, dtype=float)
+
+        for i, circ in enumerate(experiment_data._data):
+            delays[i] = circ["metadata"]["delay"]
+            count0 = circ["counts"].get("0", 0)
+            count1 = circ["counts"].get("1", 0)
+            shots = count0 + count1
+            means[i] = count1 / shots
+            stddevs[i] = np.sqrt(means[i] * (1 - means[i]) / shots)
+            # problem for the fitter if one of the std points is
+            # exactly zero
+            if stddevs[i] == 0:
+                stddevs[i] = 1e-4
+
+        if t1_guess is None:
+            t1_guess = np.mean(delays)
+        if offset_guess is None:
+            offset_guess = means[-1]
+        if amplitude_guess is None:
+            amplitude_guess = means[0] - offset_guess
+
+        fit_out, fit_err, fit_cov, chisq = curve_fit_wrapper(
+            exp_fit_fun,
+            delays,
+            means,
+            stddevs,
+            p0=[amplitude_guess, t1_guess, offset_guess],
+        )
+
+        analysis_result = AnalysisResult(
+            {
+                "value": fit_out[1],
+                "stderr": fit_err[1],
+                "unit": experiment_data._data[0]["metadata"]["unit"],
+                "label": "T1",
+                "fit": {
+                    "params": fit_out,
+                    "stderr": fit_err,
+                    "labels": ["amplitude", "T1", "offset"],
+                    "chisq": chisq,
+                    "cov": fit_cov,
+                },
+            }
+        )
+
+        #pylint: disable = too-many-boolean-expressions
+        if (
+            abs(fit_out[0] - 1.0) < 0.1
+            and abs(fit_out[2]) < 0.1
+            and chisq < 3
+            and (fit_err[0] is None or fit_err[0] < 0.1)
+            and (fit_err[1] is None or fit_err[1] < fit_out[1])
+            and (fit_err[2] is None or fit_err[2] < 0.1)
+        ):
```

Comment:
1. Define these numbers as constants.
2. This should be a general function - not specific to T1Experiment

Reply:
Let's wait also here... this time for #19